File: airbyte-commons/src/test/java/io/airbyte/commons/text/SqlsTest.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.text;

import static io.airbyte.commons.text.Sqls.toSqlName;

import com.google.common.collect.Lists;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class SqlsTest {

  enum E1 {
    VALUE_1,
    VALUE_TWO,
    value_three,
  }

  @Test
  void testToSqlName() {
    Assertions.assertEquals("value_1", toSqlName(E1.VALUE_1));
    Assertions.assertEquals("value_two", toSqlName(E1.VALUE_TWO));
    Assertions.assertEquals("value_three", toSqlName(E1.value_three));
  }

  @Test
  void testInFragment() {
    Assertions.assertEquals("('value_two','value_three')", Sqls.toSqlInFragment(Lists.newArrayList(E1.VALUE_TWO, E1.value_three)));
  }

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/test/kotlin/io/airbyte/commons/text/SqlsTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.commons.text

import com.google.common.collect.Lists
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class SqlsTest {
    internal enum class E1 {
        VALUE_1,
        VALUE_TWO,
        value_three,
    }

    @Test
    fun testToSqlName() {
        Assertions.assertEquals("value_1", Sqls.toSqlName(E1.VALUE_1))
        Assertions.assertEquals("value_two", Sqls.toSqlName(E1.VALUE_TWO))
        Assertions.assertEquals("value_three", Sqls.toSqlName(E1.value_three))
    }

    @Test
    fun testInFragment() {
        Assertions.assertEquals(
            "('value_two','value_three')",
            Sqls.toSqlInFragment(Lists.newArrayList(E1.VALUE_TWO, E1.value_three))
        )
    }
}
